Add `exempt-issue-types` option for blocklist filter by issue type

veryfancy opened the pull request:
**Description:**

Introduce a parameter `exempt-issue-types` that ensures that issues from a list of types will not be marked stale or closed.

This is a blocklist counterpart to the allowlist option `only-issue-types` added in:

- https://github.com/actions/stale/issues/1187

**Related issue:**

- https://github.com/actions/stale/issues/1293

**Check list:**
- [x] Mark if documentation changes are required.
- [x] Mark if tests were added or updated to cover the changes.

veryfancy committed:
Add exempt-issue-types option for blocklist filter by issue type

maxwxyz commented:
@luketomlinson this is similar to #1255

Bibo-Joshi reviewed (changes requested):
Hi. As a heads-up, I see that `main.ts` was not updated, which will probably lead to the input being ignored. See #1298, were I discovered that for `only-issue-types` :)

veryfancy committed:
Add missing input reading for exempt-issue-types

Bibo-Joshi reviewed (approved):
Sweet :)

veryfancy commented:
Bump to this. 🙏

veryfancy commented:
Bump to this. 🙏

laureanobrs commented:
Bump. Currently, there’s no way to process untyped issues together with only certain typed issues. This feature would make that possible. :pray:

TNATALI commented:
Bump this ya'll. Literally we are waiting on this PR to be merged and available. Would do a ton of good for the community.

veryfancy commented:
Fixed conflict with dist.js and updated behavior to keep it from filtering pull requests (to match the fix from https://github.com/actions/stale/pull/1338).

v-chiranjib-swain commented:
@veryfancy , thanks for this update. I tested the branch locally and with real Issue Types, including its interaction with `only-issue-types`, and it works as expected.

Could we document that `exempt-issue-types` takes precedence when both lists contain the same type and add a regression test for that case? This would make the behavior clear and prevent future regressions.

veryfancy commented:
@v-chiranjib-swain Thanks for testing this. I documented the precedence and added a regression test covering a type present in both lists.
